### PR TITLE
Move WebViewDidNotLeak to UITest

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change

This test is fairly flakey on device tests.
UI tests have better retry mechanics and hopefully in general because this is an actual MAUI app it just works better overall

### CODE REVIEW

File: src/Controls/tests/TestCases.HostApp/Issues/Issue22972.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                // Assert *before* the Window is closed
+                await GarbageCollectionHelper.WaitForGC(references.ToArray());
+                _rootPage.Content = new VerticalStackLayout()
+                {
+                    new Label
+                    {
+                        Text = "Success, everything has been cleaned up",
+                        AutomationId = "Success"
+                    }
+                };
```
Comment (jonathanpeppers) [inline src/Controls/tests/TestCases.HostApp/Issues/Issue22972.cs:69]: This is just an idea to make it cleaner for future tests, but could you have methods like:
```csharp
LabelAssertionHelper.Pass(_rootPage);
// or if failed
LabelAssertionHelper.Fail(_rootPage, string.Join(", ", stillAlive));
```
Then you'd use these for every memory-related test that uses a `Label`.

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22972.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+		{
+		}
+
+		public override string Issue => "Win platform WebView cannot be release after its parent window get close";
+
+		[Test]
+		[Category(UITestCategories.WebView)]
+		public void WebViewDoesNotLeak()
+		{
+			App.WaitForElement("Success", timeout: TimeSpan.FromSeconds(10));
```
Comment (jonathanpeppers) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22972.cs:19]: If this fails here, do you see the full message on the test failure in AzDO?
```csharp
"Failed to cleanup: " + string.Join(", ", stillAlive)
```

**Review by jonathanpeppers — DISMISSED** [review]

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22972.cs (hunk 2 above)
Comment (PureWeen) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22972.cs:19]: We get a screenshot of every test that fails. I could probably pull the failure text message though.

File: src/Controls/tests/TestCases.HostApp/Issues/Issue22972.cs (hunk 1 above)
Comment (PureWeen) [reply src/Controls/tests/TestCases.HostApp/Issues/Issue22972.cs]: Alright, cleaned this one up to use an extension method

So, we can now more quickly add memory-related tests!

**Review by rmarinho — APPROVED** [review]

**Pull request merged**